### DISCUSSION TITLE
请问uwp怎么处理长列表的优化

### DISCUSSION THREAD

**Discussion by Seidko** [Richasy/Bili.Uwp#1401] [问答] [answered]
我最近想写一个基于uwp的qq客户端，但是我在处理聊天内容的时候遇到了问题，那种比较活跃的群很快就会变得十分卡顿，猜测应该是长列表的问题，想请教一下怎么优化这种情况？

**Comment by Richasy** [comment, marked as answer]
我不确定你具体的代码是啥样的，但是对于基于XAML的布局来说，长列表优化的核心就在于虚拟化。
比如使用 ListView，只要你不是在 ListView 外面再套了个 ScrollViewer，那么简单的列表虚拟化问题不大。
至于更详细的列表虚拟化文档，可以参考 [ListView 和 GridView 数据虚拟化](https://docs.microsoft.com/zh-cn/windows/uwp/debug-test-perf/listview-and-gridview-data-optimization) 和 [ListView 和 GridView UI 优化](https://docs.microsoft.com/zh-cn/windows/uwp/debug-test-perf/optimize-gridview-and-listview)

我个人是用的 ScrollViewer + ItemsRepeater，这样的组合自由度更高一些，可以实现更复杂的列表布局
